### ISSUE TITLE
attribute containing `;` breaks getCSSSelector()

### ISSUE THREAD

**Issue by danschalow** [fczbkk/css-selector-generator#650]
Finding a selector for this element causes errors:

```html
<a id="calendlyButton" class="button-primary cc--button-frostbite" width="auto" ;="" href="" onclick="Calendly.initPopupWidget({url: 'REDACTED'});return false;">Kostenlose Erstberatung anfragen</a>
```

Error
```
Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document': '[;]' is not a valid selector.
```

If you look closely, although odd, the attribute `;=""` is in the html.

**Suggestion**
Use `CSS.escape` to escape the value: https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape_static

**Comment by danschalow** [comment]
Suggestion: https://github.com/fczbkk/css-selector-generator/pull/651

**Comment by fczbkk** [comment]
Hey @danschalow, thanks for the bug report. I have published `v3.6.7`, which contains a fix for this issue.

I did not use your pull request, because I used a more generic fix that uses an existing escaping function. You can see my changes in this commit:
https://github.com/fczbkk/css-selector-generator/commit/52d159ea5f4ab5f43bd2cd73cc54b6c63dfa616f

The CSS Selector Generator should now work properly even on this abomination:
```html
<div aaa:bbb="ccc" ddd;eee="fff" :="ggg" ;="hhh" : ; />
```

**Comment by danschalow** [comment]
Good stuff! Thanks for the quick feedback
